File: CSharp/Authored/wave.cs
using System;
using System.Collections.Generic;
namespace CodeWars
{
    public class Kata
    {
        public List<string> wave(string str)
        {
            List<string> result = new List<string> { };
            char[] str2 = str.ToCharArray();
            for (int b=0; b < str2.Length; b++)
            {
                if (str2[b]!=' ')
                {
                    List<char> temp = new List<char> { };
                    for (int i=0; i < str2.Length; i++)
                    {
                        if (i == b) { temp.Add(Char.ToUpper(str2[i]));} else { temp.Add(str2[i]);}
                    }
                    result.Add(string.Join("",temp));
                }
            }
            return result;
        }
    }
}

File: Ruby/Authored/Wave.rb
def wave(str)
  result = []
  str = str.split("")
  str.each_with_index do |a,b|
    if str[b] != " "
      temp = []
      str.each_with_index{|x,i| i == b ? temp << x.upcase : temp << x}
      result << temp.join
    end
  end
  result
end

letters = "abcd efghi jklmno pqrstu vwxyz".split("")
srand
1.times do
  word = []
  (0..rand(0..200)).each do
    word << letters[rand(0..letters.length)]
  end
  word = word.join
end
